File: 507-perfect-number/507-perfect-number.c
bool checkPerfectNumber(int num){
    int r = 0;
    
    for(int i = 1; i <= num/2; ++i)
    {
        if(num % i == 0)
            r += i;
    }
    if(num == r)
        return true;
    return false;
}

File: 507-perfect-number/507-perfect-number.java
class Solution {
    public boolean checkPerfectNumber(int num) {
        if(num == 1) return false;
        
        int sum = 1;
        int i = 2;
        while(i * i <= num){
            if(num % i == 0){
                sum += i;        //add first divisor
                sum += num/i;    //add last divisor
            }
            i++;
        }
        
        return (sum == num ? true : false);
    }
}